### PR TITLE
feat(BarChart): add animation

### PR DESCRIPTION
**Pull request by franckgaudin**

### CODE REVIEW

File: packages/BarChart/src/__snapshots__/BarChart.test.tsx.snap [hunk 1]
```diff
@@ -20,7 +20,7 @@ exports[`BarChart It should render a snapshot 1`] = `
         <div
           class="ids-bar-chart__graph"
           data-value="48"
-          style="width: 100%; background: rgb(251, 199, 59);"
+          style="width: 0px; background-color: rgb(251, 199, 59);"
```
Comment (vicky-comeau) [inline packages/BarChart/src/__snapshots__/BarChart.test.tsx.snap:23]: px shouldn't be needed when 0. I'd suggest removing it.

File: packages/BarChart/src/__snapshots__/BarChart.test.tsx.snap [hunk 2]
```diff
@@ -43,7 +43,7 @@ exports[`BarChart It should render a snapshot 1`] = `
         <div
           class="ids-bar-chart__graph"
           data-value="46"
-          style="width: 95.83333333333333%; background: rgb(16, 83, 255);"
+          style="width: 0px; background-color: rgb(16, 83, 255);"
```
Comment (vicky-comeau) [inline packages/BarChart/src/__snapshots__/BarChart.test.tsx.snap:46]: same as above

File: packages/BarChart/src/__snapshots__/BarChart.test.tsx.snap [hunk 3]
```diff
@@ -66,7 +66,7 @@ exports[`BarChart It should render a snapshot 1`] = `
         <div
           class="ids-bar-chart__graph"
           data-value="24"
-          style="width: 50%; background: rgb(231, 145, 154);"
+          style="width: 0px; background-color: rgb(231, 145, 154);"
```
Comment (vicky-comeau) [inline packages/BarChart/src/__snapshots__/BarChart.test.tsx.snap:69]: here too

**Comment by vicky-comeau** [comment]
I also noticed that if you change any props in the dataSet, it will reload the whole chart and the animation will run again. Is there a way that the animation will only run what the value changes? To go even deeper, it could run only for the specific bar that the value was changed and not always from 0. If the value was 50 and you set it to 75, then it should animate from 50 to 75.

File: packages/BarChart/src/__snapshots__/BarChart.test.tsx.snap (hunk 1 above)
Comment (franckgaudin) [reply packages/BarChart/src/__snapshots__/BarChart.test.tsx.snap:23]: the value is setter by `react-spring`

File: packages/BarChart/src/bar-chart.scss [hunk 4]
```diff
@@ -52,12 +53,8 @@
 
   &__graph {
     height: var(--ids-bar-chart-graph-height);
-    width: var(--ids-bar-chart-graph-width, 100%);
+    width: var(--ids-bar-chart-graph-width, var(--ids-bar-chart-graph-width));
     border-radius: var(--ids-bar-chart-graph-radius);
     background: var(--ids-bar-chart-graph-background);
-
-    &[data-value='0'] {
-      --ids-bar-chart-graph-width: #{tokens.$space-1};
```
[the remark this answers is not in the document]
Comment (franckgaudin) [reply packages/BarChart/src/bar-chart.scss:60]: errors in the dev 🤯

File: packages/BarChart/src/__snapshots__/BarChart.test.tsx.snap (hunk 1 above)
Comment (vicky-comeau) [reply packages/BarChart/src/__snapshots__/BarChart.test.tsx.snap:23]: oh right, it's the snapshot, sorry

**Comment by franckgaudin** [comment]
> I also noticed that if you change any props in the dataSet, it will reload the whole chart and the animation will run again. Is there a way that the animation will only run what the value changes? To go even deeper, it could run only for the specific bar that the value was changed and not always from 0. If the value was 50 and you set it to 75, then it should animate from 50 to 75.


https://user-images.githubusercontent.com/2379339/205365637-08ce452d-1be0-4dab-8a2b-3c8362c3b5b0.mov

File: packages/Tag/src/Tag.test.tsx [hunk 5]
```diff
@@ -132,7 +132,6 @@ describe('Tag', () => {
       src: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQEB3tCC4oJsa4ZZsiEDIhWi06EyN3iHYMoOg&usqp=CAU',
     });
     const icon = container.querySelector('.ids-tag__visual');
-    console.log(icon);
```
Comment (vicky-comeau) [inline packages/Tag/src/Tag.test.tsx:135]: shoot was that something I left? ;/

Comment (franckgaudin) [reply packages/Tag/src/Tag.test.tsx:135]: it's by chance that I saw this

**Comment by vicky-comeau** [comment]
> > I also noticed that if you change any props in the dataSet, it will reload the whole chart and the animation will run again. Is there a way that the animation will only run what the value changes? To go even deeper, it could run only for the specific bar that the value was changed and not always from 0. If the value was 50 and you set it to 75, then it should animate from 50 to 75.
> 
>  barChart-animation.mov

Looks good. I'm guessing you had to edit it with your own component because storybook just always re-renders when a property is changed?

File: packages/BarChart/src/BarChart.tsx [hunk 6]
```diff
@@ -26,9 +30,13 @@ const BarChart: React.FunctionComponent<BarChartProps> = (
...
-  if (!dataSet) {
-    return null;
-  }
+  const [animation, setAnimation] = React.useState(false);
+
+  React.useEffect(() => {
+    setAnimation(!animation);
+    const timer = setTimeout(() => setAnimation(false), 300);
+    return () => clearTimeout(timer);
+  }, []);
```
Comment (ofrogon) [inline packages/BarChart/src/BarChart.tsx:39]: `animation` should be a dependency.

Comment (ofrogon) [reply packages/BarChart/src/BarChart.tsx:39]:
```suggestion
  }, [animation]);
```

Comment (franckgaudin) [reply packages/BarChart/src/BarChart.tsx:39]: Adding `animation` in dependency will create an infinite loop, the case of figure which I wish here it is to change that value at the time when the component is mounted only.

**Comment by franckgaudin** [comment]
> Looks good. I'm guessing you had to edit it with your own component because storybook just always re-renders when a property is changed?

in fact it was easier for me to test it this way 😄

File: packages/BarChart/src/BarChart.tsx (hunk 6 above)
Comment (vicky-comeau) [reply packages/BarChart/src/BarChart.tsx:39]: ya the warnings we have for the dependency arrays are annoying :( sometimes you just want it to be empty to only run on mount! but it doesn't like that :(

**Review by ofrogon — APPROVED** [review]

**Pull request merged**